### PR TITLE
Home work24.1

### PR DESCRIPTION
**Pull request by KostiantynVorobiov**

### CODE REVIEW

File: src/main/java/com/internet/shop/Application.java [hunk 1]
```diff
@@ -27,17 +34,77 @@ public static void main(String[] args) {
...
+        shoppingCartService.addProduct(shoppingCartMcClane, nokia);
+        shoppingCartService.addProduct(shoppingCartMcClane, motorola);
+        shoppingCartService.addProduct(shoppingCartMcClane, huawei);
+        System.out.println("Alkapone's card: " + shoppingCartService.getByUserId(alkapone.getId()));
+        System.out.println("McClane's card: " + shoppingCartService.getByUserId(mcClane.getId()));
+        shoppingCartService.deleteProduct(shoppingCartMcClane, motorola);
+        System.out.println("McClane's card after delete product: "
+                + shoppingCartService.getByUserId(mcClane.getId()));
+        shoppingCartService.clear(shoppingCartMcClane);
+        System.out.println("McClane's card after clear: "
```
Comment (Vosker) [inline src/main/java/com/internet/shop/Application.java]: cart, not card)

File: src/main/java/com/internet/shop/service/ShoppingCartService.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+public interface ShoppingCartService {
+
+    ShoppingCart create(ShoppingCart shoppingCart);
+
+    ShoppingCart addProduct(ShoppingCart shoppingCart, Product product);
+
+    boolean deleteProduct(ShoppingCart shoppingCart, Product product);
+
+    void clear(ShoppingCart shoppingCart); //remove all products from the shoppingCart
```
Comment (Vosker) [inline src/main/java/com/internet/shop/service/ShoppingCartService.java]: remove comment

**Review by Vosker — COMMENTED** [review]
Good, just refactor a bit)

File: src/main/java/com/internet/shop/Application.java (hunk 1 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/Application.java]: Thanks))

File: src/main/java/com/internet/shop/service/ShoppingCartService.java (hunk 2 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/service/ShoppingCartService.java]: okey!))

File: src/main/java/com/internet/shop/dao/OrderDao.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import java.util.Optional;
+
+public interface OrderDao {
+    Order create(Order order);
+
+    Optional<Order> getOrderById(Long id);
+
+    List<Order> getAll();
+
+    List<Order> getOrderByUserId(Long userId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/OrderDao.java:14]:
```suggestion
    List<Order> getUserOrders(Long userId);
```
use plural in naming, you returns several orders, not one

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.getProducts().clear();
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return shoppingCartDao.getShoppingCardById(userId).get();
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: you just passing userId into method getShoppingCardById(Long shoppingCartId)
let's fix this

File: src/main/java/com/internet/shop/dao/ShoppingCartDao.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.model.ShoppingCart;
+import java.util.List;
+import java.util.Optional;
+
+public interface ShoppingCartDao {
+    ShoppingCart create(ShoppingCart shoppingCart);
+
+    Optional<ShoppingCart> getShoppingCardById(Long id);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/ShoppingCartDao.java]:
```suggestion
    Optional<ShoppingCart> getShoppingCartById(Long id);
```

File: src/main/java/com/internet/shop/db/Storage.java [hunk 6]
```diff
@@ -1,16 +1,43 @@
...
 
     public static void addProduct(Product product) {
         productId++;
         product.setId(productId);
         products.add(product);
     }
+
+    public static void addOrder(Order order) {
+        orderId++;
+        order.setId(orderId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/db/Storage.java]:
```suggestion
        order.setId(++orderId);
```
same in other methods

File: src/main/java/com/internet/shop/dao/OrderDao.java (hunk 3 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/dao/OrderDao.java:14]: Done

File: src/main/java/com/internet/shop/dao/ShoppingCartDao.java (hunk 5 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/dao/ShoppingCartDao.java]: Done

File: src/main/java/com/internet/shop/db/Storage.java (hunk 6 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/db/Storage.java]: Done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 4 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: Done

File: src/main/java/com/internet/shop/dao/OrderDao.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.model.Order;
+import java.util.List;
+import java.util.Optional;
+
+public interface OrderDao {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/OrderDao.java:7]: you don't need two methods getUserOrders and getOrderByUserId here;
Leave only     List<Order> getUserOrders(Long userId);

File: src/main/java/com/internet/shop/dao/OrderDao.java [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    Order create(Order order);
+
+    Optional<Order> getOrdersById(Long orderId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/OrderDao.java:10]:
```suggestion
    Optional<Order> getOrderById(Long orderId);
```

File: src/main/java/com/internet/shop/dao/OrderDao.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+    List<Order> getAll();
+
+    List<Order> getOrderByUserId(Long userId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/OrderDao.java:14]:
```suggestion
    List<Order> getUserOrders(Long userId);
```

**Review by RomanDubovskyi — APPROVED** [review]
left correct naming in comments, fix it before merging PR

**Pull request merged**